aten/src/ATen/core/ivalue.h

```diff
@@ -921,7 +913,13 @@ struct TORCH_API IValue final {
       const IValue& v);
 
   bool isPtrType() const {
-    return (isTensor() && payload.as_tensor.defined()) || is_intrusive_ptr;
+    if (isTensor()) {
+      return payload.as_tensor.defined();
+    }
+    if (isStorage() || isGenerator()) {
+      return payload.u.as_intrusive_ptr != c10::UndefinedTensorImpl::singleton();
+    }
+    return isIntrusivePtr();
```

Comment:
I presume this is for BC, so wouldn't it make sense to use `isIntrusivePtrLegacyBehavior()`?

Reply:
nice catch!